Update README.md (#80)
JuliaReach/RangeEnclosures.jl: README.md

[![Join the chat at https://gitter.im/JuliaReach/Lobby](https://badges.gitter.im/JuliaReach/Lobby.svg)](https://gitter.im/JuliaReach/Lobby?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)

A [Julia](http://julialang.org) package to bound the range of real-valued functions.

## :warning: NOTE: This library is currently a work-in-progress

## Installing

From the Julia REPL type

```julia
julia> using Pkg; Pkg.add(url="https://github.com/JuliaReach/RangeEnclosures.jl.git")
```

## Quickstart

An *enclosure* of the [range](https://en.wikipedia.org/wiki/Range_(mathematics)) of a function `f : dom ⊂ R^n -> R` is an interval
that contains the global minimum and maximum of `f` over its domain `dom`. `RangeEnclosures` offers an API to easily bound the range of
`f` with different algorithms. Here is a quick example (check the [docs](http://juliareach.github.io/RangeEnclosures.jl/latest/) for more).

```julia
julia> f(x) = -x^3/6 + 5x

julia> dom = 1 .. 4
[1, 4]

julia> enclose(f, dom, BranchAndBoundEnclosure())
[4.83333, 10.5709]
```

## Contributing

> If you encounter bugs, want to suggest new features or have questions, feel free to [open an issue](https://github.com/JuliaReach/RangeEnclosures.jl/issues/new). You can also chat with the package developers on [gitter](https://gitter.im/JuliaReach/Lobby?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge) or in the `#reachability" stream on [julia zulip](https://julialang.zulipchat.com/). Pull requests implementing new features or fixing bugs are also welcome. Make sure to checkout the [contribution guidelines](https://juliareach.github.io/RangeEnclosures.jl/dev/about/#Contributing-1).

Reviewer: ```suggestion
If you encounter bugs, want to suggest new features or have questions, feel free to [open an issue](https://github.com/JuliaReach/RangeEnclosures.jl/issues/new). You can also chat with the package developers on [gitter](https://gitter.im/JuliaReach/Lobby?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge) or in the `#reachability" stream on [julia zulip](https://julialang.zulipchat.com/). Pull requests implementing new features or fixing bugs are also welcome. Make sure to check out the [contribution guidelines](https://juliareach.github.io/RangeEnclosures.jl/dev/about/#Contributing-1).
```

Author: TIL

## Resources

- [Docs latest](http://juliareach.github.io/RangeEnclosures.jl/latest/) -- documentation of the latest stable release
- [Docs dev](http://juliareach.github.io/RangeEnclosures.jl/dev/) -- documentation of the dev version of the package
- [Contributing](https://juliareach.github.io/RangeEnclosures.jl/latest/about/#Contributing-1)
- [Release notes of tagged versions](https://github.com/JuliaReach/RangeEnclosures.jl/releases)
- [Release notes of the development version](https://github.com/JuliaReach/RangeEnclosures.jl/wiki/Release-log-tracker)

## Installing

From the Julia REPL type

```julia
julia> using Pkg; Pkg.add(url="https://github.com/JuliaReach/RangeEnclosures.jl.git")
```

## Quickstart

An *enclosure* of the [range](https://en.wikipedia.org/wiki/Range_(mathematics)) of a function `f : dom ⊂ R^n -> R` is an interval
that contains the global minimum and maximum of `f` over its domain `dom`. `RangeEnclosures` offers an API to easily bound the range of

> a function with different algorithms. Here is a quick example, check the [docs](http://juliareach.github.io/RangeEnclosures.jl/latest/) for more.

Reviewer: ```suggestion
`f` with different algorithms. Here is a quick example (check the [docs](http://juliareach.github.io/RangeEnclosures.jl/latest/) for more).
```

## Resources

- [Docs latest](http://juliareach.github.io/RangeEnclosures.jl/latest/) -- documentation of the latest stable release
- [Docs dev](http://juliareach.github.io/RangeEnclosures.jl/dev/) -- documentation of the dev version of the package
- [Contributing](https://juliareach.github.io/RangeEnclosures.jl/latest/about/#Contributing-1)
- [Release notes of tagged versions](https://github.com/JuliaReach/RangeEnclosures.jl/releases)

> - [Release notes of the development version](https://github.com/JuliaReach/RangeEnclosures.jl/wiki/Release-log-tracker)

Reviewer: ```suggestion
```

Reviewer: I am actually not so sure about this whole block. The docs are available as badges and the releases have their own standard button on GitHub. I would rather remove it completely.

Author: Agreed, there's already the releases view in github that acts as a changelog (do you keep a separated changelog in other juliareach packages?). Docs are covered by the badges. If we want something about contributing, it should be its own section in the readme instead of a link under resources?

Author: disclaimer, I am sometimes over enthusiastic about badges in readme. That being said, what do you think of a `releases-log` badge in the readme that points to gh releases page?

Author: https://github.com/juliaintervals/intervallinearalgebra.jl here's ILA readme, to show how I'm overenthusiastic about badges :)

Reviewer: The release badge in ILA is not very useful IMO because clicking it leads to nowhere and instead you can click on Releases on the right. Normal GitHub users should be familiar with that.
By the way, "latest" and "dev" docs in `RangeEnclosures` are the same. Did you mean "stable"? If you go to the docs, there is only one mode available. So you should remove that duplicate badge again.
Sometimes less is more :)

Author: The idea was to have it already ready for when the package is released, I thought `latest` meant "latest released version, dev if no release available", but I am not sure... I think "stable" is the correct way. Should we have it already there or comment out for now and reintroduce when the package is actually released?

Author: and yes the released badge in ILA only shows the current latest version, but points nowhere, could point to the releases page. Tbh the only purposes of that are 1) having 2 badges per column 2) having green-yellow somewhere, which are the colors of my [hockey team](https://en.wikipedia.org/wiki/Ilves) 😄 🏒

Reviewer: Yes, looking at other packages, the correct names are "dev" and "stable". So either change it to "stable" or remove it completely. I do not have an opinion here. In doubt, one badge seems enough and if you know Julia documentation, you can manually switch to the other version within the docs.

Author: for now I've commented the badge out, after the package is released we can decide whether we want both links or just one, I agree both can be redundant, especially if we tag a new release after every change (bug fix, new feature, etc.) in that case the two versions will differ only for 15 minutes

Reviewer: > for now I've commented the badge out

Did you push?

Author: Yes 🤔
